Fix/misc k8s fixes

madmatah opened the pull request:
## Purpose

This PR is intended to fix 5 issues : 

1. The `postgresql` Deployment template in `Ashley` causes an error in non-trashable environment type.
2. There is a confict in the secret name used to store TLS certificates in blue/green applications
3. The annotation used to declare the cert-manager `Issuer` to use in invalid in all Ingresse templates
4. There is warning due to the usage of a deprecated `apiVersion` in Ingresses templates
5. The `postgresql` Deployment template in `Richie` causes an error in non-trashable environment type.

## Proposal

1. [x] Fix Deployment template for ashley/postgresql
2. [x] Use the prefix in the secret name to have a distinct secret per Ingress
3. [x] Use `cert-manager.io/issuer` annotation instead of `cert-manager.io/cluster-issuer`
4. Upgrade all Ingresses :
   - [x] Change apiVersion `extensions/v1beta1` :arrow_right:  `networking.k8s.io/v1`
   - [x] Add [pathType](https://kubernetes.io/docs/concepts/services-networking/ingress/#path-types) argument that is now required 
   - [x] Update the backend definition according to the new spec
5. [x] Fix Deployment template for richie/postgresql

madmatah committed:
🐛(ashley) fix postgresql deployment template

When we deploy Ashley on a non-trashable environment type, the
Deployment template of the postgresql service should be empty. But due
to a bad placement of the securityContext block, an invalid k8s entity
is generated instead.

madmatah committed:
🐛(apps) fix TLS secret name in Ingresses of blue/green compatible apps

The secret name declared in the Ingress templates of blue/green
applications is a fixed string. As a result, all Ingresses generated
with the same template are using a single secret. And that causes
conflict.

madmatah committed:
🐛(apps) fix cert-manager annotation in Ingress templates of all apps

We are using a different Issuer per Namespace to generate TLS
certificates. We have to use the `cert-manager.io/issuer` annotation
instead of `cert-manager.io/cluster-issuer`, otherwise it causes an
error.

madmatah committed:
🚨(apps) fix apiVersion of Ingresses to remove a deprecation warning

When we deploy an Ingress, we have the following warning : `Warning:
extensions/v1beta1 Ingress is deprecated in v1.14+, unavailable in
v1.22+; use networking.k8s.io/v1 Ingress`

All Ingresses' apiVersion have been updated accordingly. The new
apiVersion requires to specify a pathType for each path declared in an
Ingress. And the backend declaration structure has changed.

jmaupetit commented on apps/ashley/templates/services/postgresql/deploy.yml.j2:
```diff
@@ -44,7 +44,7 @@ spec:
       securityContext:
         runAsUser: {{ container_uid }}
         runAsGroup: {{ container_gid }}
+{% endif%}
```
Is this the only impacted template?

madmatah replied:
Yes

madmatah replied:
Uh, oh. I was sure I checked everything.
But I double checked and apparently my memories are not reliable :sweat_smile: 
I'll fix this.

madmatah replied:
Done in 3d294c4b469b6997bb02bc441159dedbcc56b5ca

jmaupetit commented on apps/ashley/templates/services/nginx/ingress.yml.j2:
```diff
@@ -12,7 +12,7 @@ metadata:
     route_target_service: "app"
   annotations:
-    cert-manager.io/cluster-issuer: "{{ acme_issuer_name }}"
+    cert-manager.io/issuer: "{{ acme_issuer_name }}"
```
Should it be configurable?

madmatah replied:
Since arnold explicitly creates an Issuer per namespace during the project bootstrap, I don't think this is necessary in the current state of the project.

Should it be configurable per project, or across all arnold apps ?
It might be better to develop this in another PR, if we want to do it.

jmaupetit replied:
ok then

madmatah committed:
🐛(richie) fix postgresql deployment template

When we deploy Richie on a non-trashable environment type, the
Deployment template of the postgresql service should be empty. But due
to a bad placement of the securityContext block, an invalid k8s entity
is generated instead.
